### PR TITLE
prevent assert location during onboarding for owned hotspots

### LINKED ISSUE #335: Missing check for Hotspot Ownership
Prerequisites:
Sign into a wallet and make sure the Hotspot you pair with is not yours (typical host/owner relationship)

Bug:

Old App: Hosts use the onboarding flow to update wi-fi when they receive a Hotspot. This is a workaround since they cannot update wi-fi any other way. After connecting it to wi-fi, Hosts are met with a screen saying "you do not own this hotspot" with a button to exit setup. This terminates the onboarding flow but the wi-fi is updated.

New App: Hosts try to do the same thing, but after setting up wi-fi, they're asked to assert location, and the assert location will eventually fail (As expected). We need to bring back the Hotspot Ownership check screen and give hosts a way to exit the onboarding flow. This will appear after the "connecting to network" animation.

image

### PR DESCRIPTION
**Pull request by tyler-whitman**
closes #335 

Two new screens after selecting wifi during onboarding if the hotspot has already been onboarded.

Screen Shot 2021-03-02 at 3 53 01 PM

### CODE REVIEW

File: src/features/hotspots/setup/HotspotSetupPickWifiScreen.tsx [hunk 1]
```diff
@@ -56,12 +61,28 @@ type Route = RouteProp<HotspotSetupStackParamList, 'HotspotSetupPickWifiScreen'>
...
   const navigation = useNavigation<HotspotSetupNavigationProp>()
+  const { connectedHotspot } = useSelector((state: RootState) => state)
   const {
     params: { networks, connectedNetworks },
   } = useRoute<Route>()
+  const { showOKAlert } = useAlert()
 
-  const navSkip = () => {
-    navigation.navigate('HotspotSetupLocationInfoScreen')
+  const navSkip = async () => {
```
Comment (matthewcarlreetz) [inline src/features/hotspots/setup/HotspotSetupPickWifiScreen.tsx:70]: could `useCallback` here

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx [hunk 2]
```diff
@@ -35,6 +40,24 @@ const HotspotSetupWifiConnectingScreen = () => {
 
   const { showOKAlert } = useAlert()
 
+  const goToNextStep = async () => {
```
Comment (matthewcarlreetz) [inline src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx:43]: `useCallback`

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx [hunk 3]
```diff
@@ -35,6 +40,24 @@ const HotspotSetupWifiConnectingScreen = () => {
...
+      if (hotspot && hotspot.owner === address) {
+        navigation.replace('OwnedHotspotErrorScreen')
+      } else if (hotspot && hotspot.owner !== address) {
+        navigation.replace('NotHotspotOwnerErrorScreen')
+      } else {
+        navigation.replace('HotspotSetupLocationInfoScreen')
+      }
+    } else {
+      Logger.error('no connectedHotspot address after connecting to wifi')
+      showOKAlert({ titleKey: 'something went wrong' })
```
Comment (matthewcarlreetz) [inline src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx:56]: should we show a more informative error?

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx [hunk 4]
```diff
@@ -46,7 +69,7 @@ const HotspotSetupWifiConnectingScreen = () => {
         showOKAlert({ titleKey: 'Your password is invalid' })
```
Comment (matthewcarlreetz) [inline src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx:69]: I think this key should be `{ titleKey: 'generic.invalid_password' }`

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx [hunk 5]
```diff
@@ -35,6 +40,24 @@ const HotspotSetupWifiConnectingScreen = () => {
...
+        navigation.replace('HotspotSetupLocationInfoScreen')
+      }
+    } else {
+      Logger.error('no connectedHotspot address after connecting to wifi')
+      showOKAlert({ titleKey: 'something went wrong' })
+      navigation.goBack()
+    }
+  }
+
   const connectToWifi = () => {
```
Comment (matthewcarlreetz) [inline src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx:61]: could add a usecallback here too

**Review by matthewcarlreetz — APPROVED** [review]

**Comment by matthewcarlreetz** [comment]
Appears to work well 👍

File: src/features/hotspots/setup/HotspotSetupPickWifiScreen.tsx (hunk 1 above)
Comment (tyler-whitman) [reply src/features/hotspots/setup/HotspotSetupPickWifiScreen.tsx:70]: updated

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx (hunk 2 above)
Comment (tyler-whitman) [reply src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx:51]: updated

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx (hunk 4 above)
Comment (tyler-whitman) [reply src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx]: Updated

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx (hunk 3 above)
Comment (tyler-whitman) [reply src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx:64]: Made it `There was an error connecting to the Hotspot. Please try again.`

File: src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx (hunk 5 above)
Comment (tyler-whitman) [reply src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx:69]: updated

**Pull request merged**